server/src/test/java/io/crate/execution/engine/collect/files/FileReadingIteratorTest.java

```diff
@@ -232,6 +234,85 @@ public String readLine() throws IOException {
         tester.verifyResultAndEdgeCaseBehaviour(expectedResult);
     }
 
+    @Test
+    public void test_socket_timeouts_handled_by_retries() throws Exception {
+        ArrayList<String> fileUris = new ArrayList<>();
+
+        var tempFilePath1 = createTempFile("tempfile1", ".csv");
+        try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(tempFilePath1.toFile()),
+                                                                StandardCharsets.UTF_8)) {
+            writer.write("name,id,age\n");
+            writer.write("Arthur1,4,38\nArthur2,4,38\nArthur3,4,38\nArthur4,4,38\nArthur5,4,38\n");
+        }
+        fileUris.add(tempFilePath1.toUri().toString());
+
+        var tempFilePath2 = createTempFile("tempfile2", ".csv");
+        try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(tempFilePath2.toFile()),
+                                                                StandardCharsets.UTF_8)) {
+            writer.write("name,id,age\n");
+            writer.write("Trillian1,5,33\nTrillian2,5,33\nTrillian3,5,33\nTrillian4,5,33\nTrillian5,5,33\n");
+        }
+        fileUris.add(tempFilePath2.toUri().toString());
+
+        Reference raw = createReference("_raw", DataTypes.STRING);
+        InputFactory.Context<LineCollectorExpression<?>> ctx =
+            inputFactory.ctxForRefs(TXN_CTX, FileLineReferenceResolver::getImplementation);
+        List<Input<?>> inputs = Collections.singletonList(ctx.add(raw));
+
+
+        Supplier<BatchIterator<Row>> batchIteratorSupplier =
+            () -> new FileReadingIterator(
+                fileUris,
+                inputs,
+                ctx.expressions(),
+                null,
+                Map.of(LocalFsFileInputFactory.NAME, new LocalFsFileInputFactory()),
+                false,
+                1,
+                0,
+                List.of("name", "id", "age"),
+                new CopyFromParserProperties(true, true, ','),
+                CSV,
+                Settings.EMPTY
+            ) {
+                int retry = 0;
+
+                @Override
+                BufferedReader createBufferedReader(InputStream inputStream) throws IOException {
+                    return new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8)) {
+
+                        private int currentLineNumber = 0;
+
+                        @Override
+                        public String readLine() throws IOException {
+                            var line = super.readLine();
+                            if (new Random().nextBoolean()) {
+                                // current implementation does not handle SocketTimeoutException thrown when parsing header so skip it here as well.
+                                if (currentLineNumber++ > 0 &&
+                                    new Random().nextBoolean() && retry++ < MAX_SOCKET_TIMEOUT_RETRIES) {
```

Comment:
Lets make the test deterministic
```suggestion
                                    retry++ < MAX_SOCKET_TIMEOUT_RETRIES) {
```

Reply:
missed this, will take care of it.